integration-cli/docker_cli_build_test.go

```diff
@@ -5444,3 +5445,513 @@ func (s *DockerTrustSuite) TestBuildContextDirIsSymlink(c *check.C) {
+	envKey := "foo"
+	envVal := "bar"
+	envKey1 := "foo1"
+	envValOveride := "barOverride"
+	buildCmd := exec.Command(dockerBinary, "build", "-t", imgName, "--build-arg",
+		fmt.Sprintf("%s=%s", envKey, envVal), "-")
+	buildCmd.Stdin = strings.NewReader(fmt.Sprintf(`FROM busybox
+		ARG %s
+		ENV %s %s
+		ENV %s ${%s}
+		RUN echo $%s
+		CMD echo $%s`, envKey, envKey, envValOveride, envKey1, envKey, envKey1, envKey1))
+
+	if out, _, err := runCommandWithOutput(buildCmd); err != nil || strings.Count(out, envValOveride) != 2 {
+		if err != nil {
+			c.Fatalf("build failed to complete: %v %v", out, err)
+		}
+		c.Fatalf("failed to access environment variable in output: '%v' expected: '%v'", out, envValOveride)
+	}
+
+	containerName := "bldargCont"
+	runCmd := exec.Command(dockerBinary, "run", "--name", containerName, imgName)
+	if out, _, err := runCommandWithOutput(runCmd); !strings.Contains(out, envValOveride) || err != nil {
+		c.Fatalf("run produced invalid output: %q, expected %q", out, envValOveride)
+	}
+}
+
+func (s *DockerSuite) TestBuildBuildTimeArgUntrustedDefinedAfterUse(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "foo"
+	envVal := "bar"
+	buildCmd := exec.Command(dockerBinary, "build", "-t", imgName, "--build-arg",
+		fmt.Sprintf("%s=%s", envKey, envVal), "-")
+	buildCmd.Stdin = strings.NewReader(fmt.Sprintf(`FROM busybox
+		RUN echo $%s
+		ARG %s
+		CMD echo $%s`, envKey, envKey, envKey))
+
+	if out, _, err := runCommandWithOutput(buildCmd); err != nil || strings.Contains(out, envVal) {
+		if err != nil {
+			c.Fatalf("build failed to complete: %v %v", out, err)
+		}
+		c.Fatalf("able to access environment variable in output: '%v' expected to be missing", out)
+	}
+
+	containerName := "bldargCont"
+	runCmd := exec.Command(dockerBinary, "run", "--name", containerName, imgName)
+	if out, _, err := runCommandWithOutput(runCmd); out != "\n" || err != nil {
+		c.Fatalf("run produced invalid output: %q, expected empty string", out)
+	}
+}
+
+func (s *DockerSuite) TestBuildBuildTimeArgBuiltinArg(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "HTTP_PROXY"
+	envVal := "bar"
+	buildCmd := exec.Command(dockerBinary, "build", "-t", imgName, "--build-arg",
+		fmt.Sprintf("%s=%s", envKey, envVal), "-")
+	buildCmd.Stdin = strings.NewReader(fmt.Sprintf(`FROM busybox
+		RUN echo $%s
+		CMD echo $%s`, envKey, envKey))
+
+	if out, _, err := runCommandWithOutput(buildCmd); err != nil || !strings.Contains(out, envVal) {
+		if err != nil {
+			c.Fatalf("build failed to complete: %v %v", out, err)
+		}
+		c.Fatalf("failed to access environment variable in output: '%v' expected: '%v'", out, envVal)
+	}
+
+	containerName := "bldargCont"
+	runCmd := exec.Command(dockerBinary, "run", "--name", containerName, imgName)
+	if out, _, err := runCommandWithOutput(runCmd); out != "\n" || err != nil {
+		c.Fatalf("run produced invalid output: %q, expected empty string", out)
+	}
+}
+
+func (s *DockerSuite) TestBuildBuildTimeArgLogging(c *check.C) {
```

Comment:
Not sur I understand what this test is doing. Can you explain? What is `Logging` here?